### DISCUSSION TITLE
Get GPS location in background

### DISCUSSION THREAD

**Discussion by pieteroskam** [zepp-health/zeppos-forum#281] [Q&A] [answered]
Hi there,

I develop several GPS based apps for fitbits, Apple, Wear OS and Huawei. I want to explore the Zepp OS opportunities. Before buying the actual watch, can someone tell me if the following things are possible.

1) keep the app running in the foreground, so when the screen is turned off the app wont get killed
2) Get location updates in the background so we can plot a route or determine if we should continue to the next waypoint
3) store (json) object in memory and load them later.
4) (OPTIONAL) show a interactive map (like google maps or other provider)
5) Force use the watch internal GPS instead of the Phone.

**Comment by SashaCX75** [comment, marked as answer]
I think you need to decide which version of ZeppOS you are targeting. The available features depend on it.

You can read about background services here
https://docs.zepp.com/docs/guides/framework/device/app-service/
You can also make the programme not run in the background, but continue to run after the screen is switched on.

You can read about how to store a string in memory here
https://docs.zepp.com/docs/reference/device-app-api/newAPI/fs/writeFileSync/
or here
https://docs.zepp.com/docs/reference/device-app-api/newAPI/storage/localStorage/

You can read about using geolocation on your watch here
https://docs.zepp.com/docs/reference/device-app-api/newAPI/sensor/Geolocation/

**Reply by MyoungXUE** [reply]
Hello @SashaCX75 , thank you for your answer, very professional!

**Reply by pieteroskam** [reply]
@SashaCX75 Thanks a lot!!

I guess i will then start some developing and experimenting today. 

If you have any interest in co-developing this, let me know. My app generates quitte some revenue on the other platforms. We can discuss a co-revenue model. Though I realy like programming, maintaining 6 code bases is sometimes quite challenging. Now i have yet to learn another IDE / ecosystem.

**Comment by MyoungXUE** [comment]
Hello @pieteroskam and welcome to the Zepp OS community!

I will answer your questions one by one.

1. Keep running in the foreground, but keep the screen always on, so you don't have to worry about power consumption, which is the strength of Amazfit watches.
2. Our Mini Program can support running in the background, but cannot call GPS related interfaces in the background, please refer to https://docs.zepp.com/docs/guides/framework/device/app-service/
3. Referring to the information provided by Sasha, `writeFileSync` and `localStorage` are both available
4. There is currently no open map interface, so this implementation may be difficult
5. You can use the Geolocation interface, as Sasha said, it uses the positioning of the watch.

**Reply by pieteroskam** [reply]
@MyoungXUE 

Thanks. also thank you for complementing the answers by @SashaCX75 

It is a good sign that my first question in the community is answered so quickly and complete!

**Reply by MyoungXUE** [reply]
Zepp OS official technicians are active in the community and can usually get a reply within 24 hours.

We believe that Zepp OS is community building, and if you have any good ideas, you can let us know.

Regarding the GPS issue, may I ask what functions the application you want to develop has? Can it be experienced in the Compete you mentioned? We are also considering improving support for GPS and hope that the platform can have more high-quality GPS applications.

**Reply by pieteroskam** [reply]
hi @MyoungXUE 

The app that i want to make first is for golfing. See examples here
https://apps.apple.com/nl/app/golf4watch-rangefinder/id1584652910
https://gallery.fitbit.com/details/859975b5-8dcd-4d98-ab1c-1592f5f3c878
